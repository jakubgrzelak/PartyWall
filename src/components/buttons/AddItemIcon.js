import React from 'react';
import {
  View,
  StyleSheet,
  Dimensions,
} from 'react-native';
import { navigate } from '../../navigators/navigationActions';
import { Icon } from 'react-native-elements'
import Colors from '../../themes/colors';

const screenWidth = Dimensions.get('screen').width;
// const addIcon = require('../../assets/nav/add_icon.png');

export const AddItemIcon = (props) => {
  function onIconPress() {
    navigate('AddItemScreen');
  }

  return (
    <View style={styles.container}>
      <Icon
        name='plus-circle'
        type='feather'
        color={Colors.addButton}
        onPress={onIconPress}
        size={48}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: 58,
    height: 58,
    borderRadius: 30,
    backgroundColor: '#fff',
    bottom: 15,
    marginHorizontal: 10,
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    width: 52,
    height: 58,
  },
  floatContainer: {
    position: 'absolute',
    width: screenWidth - 60,
    height: 85,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
