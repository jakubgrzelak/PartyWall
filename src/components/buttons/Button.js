import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import Colors from '../../themes/colors';

export const Button = ({
  text,
  onPress,
  buttonStyles,
  textStyles,
  disabled = false,
  disabledStyle,
}) => {
  function returnButtonStyle() {
    if (disabled) {
      return [styles.container, buttonStyles, disabledStyle];
    }
    return [styles.container, buttonStyles];
  }

  function returnTextStyle() {
    return [styles.textStyles, textStyles];
  }

  return (
    <TouchableOpacity
      disabled={disabled}
      onPress={onPress}
      style={returnButtonStyle()}>
      <Text style={returnTextStyle()}>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 50,
    borderRadius: 5,
    width: '100%',
    backgroundColor: Colors.mainButtonBackground,
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 3,
  },
  textStyles: {
    color: Colors.brightText,
    fontSize: 20,
    fontWeight: '700',
  },
});
