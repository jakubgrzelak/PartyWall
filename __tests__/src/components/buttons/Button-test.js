import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { shallow } from 'enzyme';
import { Button } from '../../../../src/components/buttons/Button';

describe('Rendering', () => {
  it('should match to snapshot', () => {
    const tree = render(<Button text="Button" />).toJSON();
    expect(tree).toMatchSnapshot()
  });
});

describe('On press action', () => {
  it('should be called', () => {
    const mockPress = jest.fn();
    const { getByText } = render(<Button text="Submit" onPress={mockPress} />);

    fireEvent.press(getByText('Submit'));
    expect(mockPress).toBeCalled();
  });
})
