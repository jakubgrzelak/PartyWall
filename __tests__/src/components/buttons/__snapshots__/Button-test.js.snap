// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should match to snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#157dfb",
      "borderRadius": 5,
      "height": 50,
      "justifyContent": "center",
      "marginVertical": 3,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "700",
        },
        undefined,
      ]
    }
  >
    Button
  </Text>
</View>
`;
